internal/wasm/store.go

```diff
@@ -619,3 +626,17 @@ func (s *Store) getFunctionTypeID(t *FunctionType) (FunctionTypeID, error) {
 	}
 	return id, nil
 }
+
+func (s *Store) Close(ctx context.Context) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	var err error
+	// Close modules in reverse initialization order.
+	for i := len(s.moduleNames) - 1; i >= 0; i-- {
+		_, err = s.modules[s.moduleNames[i]].CallCtx.close(ctx, 0)
+	}
+	s.moduleNames = nil
+	s.modules = map[string]*ModuleInstance{}
+	// Only returns last error.
```

Comment:
I couldn't find a pattern for returning multiple errors so did this as an initial try. Let me know if there's a way we would want to combine the errors or go with it.

Reply:
Probably we've not been as explicit in our code here as func-e, but we should use trailing comments and return first err (vs try to combine etc).

ex. from func-e.. I'm pretty sure we do this somewhere in wazero, but didn't trailing comment
```
	var err error
	go func() {
		if e := r.Run(ctx, args); e != nil && err == nil {
			err = e // first error
		}
		ran <- true
	}()
```